core/tx_list.go

```diff
@@ -696,18 +696,19 @@ func (l *txPricedList) Underpriced(tx *types.Transaction) bool {
 }
 
 func (l *txPricedList) underpricedForMulti(h *multiCurrencyPriceHeap, tx *types.Transaction) bool {
```

Comment:
Why do we check against all price heaps and not only the heap for the fee currency of the TX? I think I'm missing the logic here.

Reply:
the underprice check tries to find out if this tx is too cheap for this pool (in its entirety), so as to drop it if no slots are available. It might not be the cheapest of its currency, but there might be another tx from a different currency that's cheaper, so it makes sense to eventually keep this one and remove a another one

Reply:
> It might not be the cheapest of its currency, but there might be another tx from a different currency that's cheaper, so it makes sense to eventually keep this one and remove a another one

Why don't we remove all underpriced transactions?

Reply:
Underpriced is a term for a new tx in the (already full) pool that has a gas price lower than all of the txs in the pool.

Reply:
Ok, so in theory we could find a transaction in a different pool that is cheaper than the one we're processing and remove that instead.

But does that make sense given that we can only add the current on to the pool matching the fee currency?

And if we want to take this path, I would expect that we need the exchange rates to convert prices between the different heaps. How can you work around that?

Reply:
Discussed offline: There's only one pool,but many heaps. Exchange rate conversion is already being done on getting the cheapest tx and also in underprice check (in this PR)